src/arch/bsdsocketinterf.cpp

```diff
@@ -239,7 +273,34 @@ CBSDSocketInterface::TSocketDescriptor CBSDSocketInterface::openUDPReceivePort(c
           //if this fails we may have given a non multicasting addr. For now we accept this. May need to be changed in the future.
           DEVLOG_WARNING("CBSDSocketInterface: setsockopt(IP_ADD_MEMBERSHIP) failed: %s\n", strerror(errno));
         }
-#endif
+#else
+        // setting up multicast group
```

Comment:
I would prefer to have that somewhere else. If that is not possible at all please move it at least into a function so that the code stays readable.

Reply:
There are #ifdefs for many archs in that file by now. Short of a full refactoring, I've done as suggested and designed a local linkage (anon namespace) function - looks pretty readable to me now, I hope it's fine with you as well. Thank you for the input.

Reply:
Yes there are. We are working on 4diac FORTE for nearly 20 years. There is still stuff in it that is not as good as it could be. Also we learned a lot in the last 20 years. For us `#ifdef`are now technical depth making maintenance harder. So if we can avoid it in new code we try as hard as possible to do so.